util/httputil/client.go

```diff
@@ -125,6 +127,21 @@ type TLSOptions struct {
 	ServerName         string
 }
 
+func TLSConfig(cfg config.TLSConfig) (*tls.Config, error) {
```

Comment:
`TLSConfig` is not great naming, since both `TLSConfig` and `NewTLSConfig` are conceptually `New...` methods.

Conceptually, it seems that this package has its own `TLSOptions` type in order to not have to know anything about Prometheus's `config` package types. Adding this helper function here changes that. Maybe this new helper function should live in another package, like `config`? Otherwise we might as well do away with the `TLSOptions` type, to which we merely copy over all the identical fields... (disclaimer: haven't looked very deeply at this).